api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -124,8 +126,25 @@ message DownstreamTlsContext {
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
 }
 
+// TLS key log configuration.
+// The key log file format is "format used by NSS for its SSLKEYLOGFILE debugging output" (text taken from openssl man page)
+message TlsKeyLog {
+  // The path to save the TLS key log.
+  string tls_keylog_path = 1;
+
+  // The src IP address that will be used to filter the connection which should save the TLS key log
+  config.core.v3.Address tls_keylog_src = 2;
+
+  // The destination IP address that will be used to filter the connection which should save the TLS key log
+  config.core.v3.Address tls_keylog_dst = 3;
```

Comment:
Given that this is set on a listener, do we need a destination address at all?

Reply:
This is in `CommonTlsContext`, which means it will also be used for upstream connections.

Reply:
Oh, good point. Ignore my comment then.